lib/octicons_gem/lib/octicons_v2/octicon_v2.rb

```diff
@@ -84,5 +81,25 @@ def calculate_width(height)
     def calculate_height(width)
       (width.to_i * @height) / @width
     end
+
+    def get_octicon(symbol, options)
+      if octicon = OcticonsV2::OCTICON_SYMBOLS[symbol]
+        height = options[:height] || options[:width] || 16
+        octicon_height = closest_octicon_height(octicon["heights"].keys, height)
+        return {
+          name: octicon["name"],
+          keywords: octicon["keywords"],
+          width: octicon["heights"][octicon_height.to_s]["width"].to_i,
+          height: octicon_height,
+          path: octicon["heights"][octicon_height.to_s]["path"]
+        }
+      end
+    end
+
+    def closest_octicon_height(octicon_heights, height)
+      return octicon_heights.reduce(octicon_heights[0].to_i) {
+        |acc, octicon_height| (octicon_height.to_i <= height.to_i ? octicon_height.to_i : acc)
+      }
```

Comment:
I would make this cleaner to read

```suggestion
      return octicon_heights.reduce(octicon_heights[0].to_i) do |acc, octicon_height| 
        octicon_height.to_i <= height.to_i ? octicon_height.to_i : acc
      end
```